packages/shared/src/hdf5-models.ts

```diff
@@ -216,7 +216,7 @@ export interface UnknownType {
 export type ScalarValue<T extends DType = DType> = T extends NumericLikeType
   ?
       | number
-      | (T extends NumericType ? bigint : never) // let providers return bigints
+      | (T extends IntegerType ? bigint : never) // let providers return bigints
```

Comment:
Unrelated. This is more precise but has no impact on the codebase since we don't have visualizations that accept only integers or only floats.